bat/LINK.BAT

```diff
@@ -11,28 +11,31 @@ if /I "%1" == "tron" (set z_TraceMode=tron
 set /A z_NestLevel=%z_NestLevel%+1
 rem ----- Lvl(%z_NestLevel%) Start %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
 
-if /I %1. == . (set /P z_file=Hit Enter for help or suppply name of obj file to link: 
+if /I %1. == . (set /P z_file=Hit Enter for help or supply name of obj file to link: 
         ) else (set z_file=%1)
 if /I %z_file%. == . goto help
 :chkfile
 rem convert Unix path notation to Windows before checking the file
 set z_file=%z_file:/=\%
-if exist %z_file%.OBJ goto file_ok
-rem if user specified valid extension, strip it off
-if /I %z_file:~-4%. NEQ .OBJ. goto help
-set z_file=%z_file:~0,-4%
-if exist %z_file%.OBJ goto file_ok
-echo %0 ERROR: %z_file%.OBJ was not found
-set z_ReturnCode=16
-goto return
+call bat\fparts.bat %z_TraceMode% %z_file%
+set z_file=%z_fdrv%%z_fpath%%z_fname%
+if exist %z_file%%z_fext% goto file_ok
+rem if extension specified: error because no default applies
+if /I %z_fext%. NEQ . goto notfound
+rem try with default extension of obj
+set z_fext=.obj
```

Comment:
Lines 25-26. obj should be OBJ. In general, filenames, including extensions, must be upper case.